api/provider/aws/load_balancer_delete.go

```diff
@@ -0,0 +1,102 @@
+package aws
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/elb"
+	"github.com/aws/aws-sdk-go/service/iam"
+)
+
+func (l *LoadBalancerProvider) Delete(loadBalancerID string) error {
+	fqLoadBalancerID := addLayer0Prefix(l.Config.Instance(), loadBalancerID)
+
+	if err := l.deleteLoadBalancer(fqLoadBalancerID); err != nil {
+		return err
+	}
+
+	roleName := fmt.Sprintf("%s-lb", fqLoadBalancerID)
+	if err := l.deleteRole(roleName); err != nil {
+		return err
+	}
+
+	// todo: do we need to delete role policy? Or does that happen in deleteRole?
+
+	securityGroupName := fmt.Sprintf("%s-lb", fqLoadBalancerID)
+	securityGroup, err := readSG(l.AWS.EC2, securityGroupName)
+	if err != nil && !strings.Contains(err.Error(), "does not exist") {
+		return err
+	}
+
+	if securityGroup != nil {
+		groupID := aws.StringValue(securityGroup.GroupId)
+		if err := deleteSG(l.AWS.EC2, groupID); err != nil {
+			return err
+		}
+	}
+
+	if err := l.deleteTags(loadBalancerID); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (l *LoadBalancerProvider) deleteLoadBalancer(loadBalancerName string) error {
+	input := &elb.DeleteLoadBalancerInput{}
+	input.SetLoadBalancerName(loadBalancerName)
+
+	if _, err := l.AWS.ELB.DeleteLoadBalancer(input); err != nil {
+		if err, ok := err.(awserr.Error); ok && err.Code() == "NoSuchEntity" {
+			return nil
+		}
+
+		return err
+	}
+
+	return nil
+}
+
+func (l *LoadBalancerProvider) deleteRole(roleName string) error {
+	input := &iam.DeleteRoleInput{}
+	input.SetRoleName(roleName)
+
+	// todo: validate NoSuchEntity is correct error code
+	if _, err := l.AWS.IAM.DeleteRole(input); err != nil {
+		if err, ok := err.(awserr.Error); ok && err.Code() == "NoSuchEntity" {
+			return nil
+		}
+
+		return err
+	}
+
+	return nil
+}
+
+func (l *LoadBalancerProvider) deleteTags(loadBalancerID string) error {
+	tags, err := l.TagStore.SelectByTypeAndID("load_balancer", loadBalancerID)
+	if err != nil {
+		return err
+	}
+
+	for _, tag := range tags {
+		if err := l.TagStore.Delete(tag.EntityType, tag.EntityID, tag.Key); err != nil {
+			return err
+		}
+	}
+
+	serviceTags, err := l.TagStore.SelectByType("service")
```

Comment:
Since this tag "belongs" to the service, I think it makes more sense to delete it with the service.